pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1724,11 +1726,15 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 						Issue in Libvirt: https://gitlab.com/libvirt/libvirt/-/issues/608
 						once the issue is resolved we can remove mpx disablement
 		*/
-		if _, exists := existingFeatures["mpx"]; !exists && vmi.Spec.Domain.CPU.Model != v1.CPUModeHostModel && vmi.Spec.Domain.CPU.Model != v1.CPUModeHostPassthrough {
-			domain.Spec.CPU.Features = append(domain.Spec.CPU.Features, api.CPUFeature{
-				Name:   "mpx",
-				Policy: "disable",
-			})
+		// skip the mpx CPU feature validation for s390x as it is not supported.
+		if !checks.IsS390X(runtime.GOARCH) {
+
+			if _, exists := existingFeatures["mpx"]; !exists && vmi.Spec.Domain.CPU.Model != v1.CPUModeHostModel && vmi.Spec.Domain.CPU.Model != v1.CPUModeHostPassthrough {
```

Comment:
Can we use a single if here

Reply:
done

Reply:
Hi @Barakmor1  I have updated the code to use single if. Please let me know if it is fine or not?
Thanks